src/App.tsx

```diff
@@ -15,36 +16,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ALPHABETICALLY = 'sort-alphabetically',
+  BY_LENGTH = 'sort-by-length',
+  NONE = '',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  let visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.ALPHABETICALLY:
+        return good1.localeCompare(good2);
+      case SortType.BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  const resetSort = () => {
+    setSortField('');
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SortType.ALPHABETICALLY,
+          })}
+          onClick={() => setSortField(SortType.ALPHABETICALLY)}
```

Comment:
[UX] - You should mark the active button to indicate to the user what action is active. You can achieve this by adding an active class to the button when its action is active.